modules/wallet/persist.go

```diff
@@ -124,6 +154,13 @@ func (w *Wallet) initPersist() error {
 	if err != nil {
 		return err
 	}
+
+	// Open the database.
+	err = w.openDB(filepath.Join(w.persistDir, dbFile))
+	if err != nil {
+		return err
+	}
```

Comment:
while we're here, it might make sense to create a threadgroup to handling the closing of the db. Maybe out of scope for this PR though.